### DISCUSSION TITLE
trouble when use finn.transformation.fold_constants

### DISCUSSION THREAD

**Discussion by codeSlayer9527** [Xilinx/finn#583] [Q&A]
I follow the 1_brevitas_network_import and try to build my own model,but when I try to run the cleanup step it reports:

`model = model.transform(FoldConstants())

Exception                                 Traceback (most recent call last)
<ipython-input-13-4d55c567f253> in <module>
----> 1 model = model.transform(FoldConstants())

/workspace/finn-base/src/finn/core/modelwrapper.py in transform(self, transformation, make_deepcopy, cleanup, fix_float64)
    139         model_was_changed = True
    140         while model_was_changed:
--> 141             (transformed_model, model_was_changed) = transformation.apply(
    142                 transformed_model
    143             )

/workspace/finn-base/src/finn/transformation/fold_constants.py in apply(self, model)
     57                 # this node has no dynamic inputs, only constant ones -- so we can
     58                 # do constant folding.
---> 59                 oxe.execute_node(n, execution_context, graph)
     60                 # use the execution result as an initializer
     61                 model.set_initializer(node_out, execution_context[node_out])

/workspace/finn-base/src/finn/core/onnx_exec.py in execute_node(node, context, graph, return_full_exec_context, opset_version)
    119                 # use that index to index output_list
    120                 if output_list[list_ind].shape != context[outp].shape:
--> 121                     raise Exception(
    122                         """Output shapes disagree after node execution:
    123                         found %s vs expected %s"""

Exception: Output shapes disagree after node execution:
                        found (4,) vs expected (0,)`

I have no idea how to solve it

**Comment by fpjentzsch** [comment]
Hi, could you share your input .onnx model? There seems to be something wrong with it.

**Reply by codeSlayer9527** [reply]
Hello fpjentzsch，

this is my .onnx model 
[unet_test2.zip](https://github.com/Xilinx/finn/files/8452811/unet_test2.zip)

It is a unet model for image segmentation, and i did a little change in its conv layer to reduce model size

When I first met the problem I try to find some solution in my input tensor because I guess the 4 means the 4-dimension input，but I didn't know what the 0 means 
![CF341A056A17F46283582E69D54FC7F3](https://user-images.githubusercontent.com/102790324/162462686-f87b5760-15eb-4b96-911e-3706c2fa2a09.jpg)
![415D360581DEF84BE19AC9439F0427BA](https://user-images.githubusercontent.com/102790324/162462705-07283a66-00b4-4eec-99ae-4092088f3ca8.jpg)

If you need my original model I'm glad to share it.

**Reply by mllearner98** [reply]
Hi, did you solve your problem. Because I have the same problem but my error is as:

> Exception: Output shapes disagree after node execution:
>                         found (1, 3, 3) vs expected ()

Please help if you have figured it out.

**Reply by mllearner98** [reply]
> 

Hi, did you solve your problem. Because I have the same problem but my error is as:

Exception: Output shapes disagree after node execution:
found (1, 3, 3) vs expected ()

Please help if you have figured it out.

**Comment by julienmas** [comment]
Hi newpythonnoob, I'm also working on implementing a U-Net with FINN. I'd be interested to discuss this with you. Here's my email address: julien.mastrangelo@polymtl.ca
